File: TiaCodegen/Commands/Functions/RTrigCall.cs
using System.Linq;
using TiaCodegen.Commands.Functions.Base;
using TiaCodegen.Enums;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions
{
    public class RTrigCall : SystemFunctionBlockCall
    {
        public RTrigCall(string instanceName,
            IOperationOrSignal clk = null,
            IOperationOrSignal q = null,
            IOperationOrSignal eno = null) : base("R_TRIG", instanceName, eno)
        {
            Interface["CLK"] = new IOperationOrSignalDirectionWrapper(clk, Direction.Input);
            Interface["Q"] = new IOperationOrSignalDirectionWrapper(q, Direction.Output);

            Children.AddRange(Interface.Values.Where(x => x.OperationOrSignal != null).Select(x => x.OperationOrSignal));
        }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/RTrigCall.ts
import { IOperationOrSignal } from '../../Interfaces/IOperationOrSignal.js';
import { IOperationOrSignalDirectionWrapper } from '../../Interfaces/IOperationOrSignalDirectionWrapper.js';
import { Direction } from '../../Enums/Direction.js';
import { SystemFunctionBlockCall } from './Base/SystemFunctionBlockCall.js';

export interface RTrigCallOptions {
    instanceName: string;
    clk?: IOperationOrSignal | null;
    q?: IOperationOrSignal | null;
    eno?: IOperationOrSignal | null;
}

export class RTrigCall extends SystemFunctionBlockCall {
    constructor(options: RTrigCallOptions) {
        const { instanceName, clk = null, q = null, eno = null } = options;
        super({ functionName: 'R_TRIG', instanceName, eno });
        this.iface['CLK'] = new IOperationOrSignalDirectionWrapper(clk, Direction.Input);
        this.iface['Q'] = new IOperationOrSignalDirectionWrapper(q, Direction.Output);

        for (const w of Object.values(this.iface)) {
            if (w.operationOrSignal !== null) this.children.push(w.operationOrSignal);
        }
    }
}
